src/aws_encryption_sdk/materials_managers/mpl/materials.py

```diff
@@ -136,3 +143,15 @@ def data_key(self) -> DataKey:
     def verification_key(self) -> bytes:
         """Materials' verification key."""
         return self.mpl_materials.verification_key
+
+    @property
+    def encryption_context(self) -> Dict[str, str]:
+        """Materials' encryption context."""
+        return self.mpl_materials.encryption_context
+
+    @property
+    # Pylint thinks this name is too long, but it's the best descriptor for this...
+    # pylint: disable=invalid-name
+    def required_encryption_context_keys(self) -> bytes:
```

Comment:
I think that this is `reproducedEncryptionContext`. The two sides are not symmetrical.

Reply:
This is how it's called in the MPL `DecryptionMaterials`: https://github.com/aws/aws-cryptographic-material-providers-library/blob/main/AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/Model/structures.smithy#L79

It is true that the `DecryptionMaterialsRequest` takes in a `reproducedEncryptionContext` argument: https://github.com/aws/aws-cryptographic-material-providers-library/blob/main/AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/Model/cmms.smithy#L108

I believe the `reproducedEncryptionContext` from the request is passed into the Dafny layer, where it is validated, then the "response" here includes this as output.